pkg/deployment/inventory.go

```diff
@@ -69,6 +71,41 @@ func GenerateRoleInventory(ctx context.Context, helper *helper.Helper,
 		}
 	}
 
+	if instance.Spec.NodeTemplate.Nova != nil {
+		if _, ok := roleNameGroup.Vars["edpm_ovn_metadata_agent_metadata_agent_DEFAULT_nova_metadata_host"]; !ok {
+			novaSvc := &corev1.Service{}
+			err = helper.GetClient().Get(ctx, types.NamespacedName{Name: "nova-metadata-internal", Namespace: instance.Namespace}, novaSvc)
```

Comment:
I'm not a huge fan of hardcoding such dependencies here. Also note that this code path will disappear as soon as https://github.com/openstack-k8s-operators/dataplane-operator/pull/337 merges